File: calendar-generator/defaults.js
const moment = require('moment');

const defaultNames = require('./default-names.json');
const processContext = require('./process-context.js');

// Set default input values, if no context is available
const defaultContext = {
  period: 'full-year',
  year: moment().year().toString(),
  'from-date': `${moment().year().toString()}-01-01`,
  'from-date-string': 'January 1st',
  'to-date': `${moment().year().toString()}-12-31`,
  'to-date-string': 'December 31st',
  order: 'descending',
  'level-for-months': 'false',
  'level-for-weeks': 'true',
  'month-language': 'english',
  'month-case': 'upper',
  'month-number': 'prefix',
  'month-number-or-year': 'month',
  'month-separator': ' ',
  'week-format': 'date',
  'week-number': 'none',
  'week-number-label': '',
  'week-date': 'prefix',
  'week-date-separator': ' ',
  'week-date-abbreviated': 'true',
  'week-date-abbreviated-length': '3',
  'day-language': 'english',
  'day-case': 'upper',
  'day-number': 'prefix',
  'day-separator': ' ',
  'day-abbreviated': 'true',
  'day-abbreviated-length': '3',
};

// Convert default month names to selected case
const defaultMonthNames = defaultNames.months[defaultContext['month-language']];
Object.keys(defaultMonthNames).forEach(function convertMonthCase(key) {
  defaultMonthNames[key] = processContext.convertCase(defaultMonthNames[key], defaultContext['month-case']);
});

// Convert default day names to selected case
const defaultDayNames = defaultNames.days[defaultContext['day-language']];
Object.keys(defaultDayNames).forEach(function convertDayCase(key) {
  defaultDayNames[key] = processContext.convertCase(defaultDayNames[key], defaultContext['day-case']);
});

// Add default month and day names to context and export
Object.assign(defaultContext, defaultMonthNames, defaultDayNames);
module.exports.context = defaultContext;

File: python/defaults.py
import datetime
import json


with open('default-names.json', 'r') as default_names_json:
    default_names = json.loads(default_names_json.read())

default_context = {
    'period': 'full year',
    'year': str(datetime.datetime.now().year),
    'from_date': '{}-01-01'.format(datetime.datetime.now().year),
    'from_date_string': 'January 1st',
    'to_date': '{}-12-31'.format(datetime.datetime.now().year),
    'to_date_string': 'December 31st',
    'order': 'descending',
    'level_for_months': 'false',
    'level_for_weeks': 'true',
    'month_language': 'english',
    'month_case': 'upper',
    'month_number': 'prefix',
    'month_number_or_year': 'month',
    'month_separator': ' ',
    'week_format': 'date',
    'week_number': 'prefix',
    'week_number_label': 'WEEK ',
    'week_date': 'prefix',
    'week_date_separator': ' ',
    'week_date_abbreviated': 'true',
    'week_date_abbreviated_length': '3',
    'day_language': 'english',
    'day_case': 'upper',
    'day_number': 'prefix',
    'day_separator': ' ',
    'day_abbreviated': 'true',
    'day_abbreviated_length': '3',
}

if __name__ == '__main__':
    print(json.dumps(default_context, indent=4))
